py/lexer.c

```diff
@@ -531,10 +531,12 @@ STATIC bool skip_whitespace(mp_lexer_t *lex, bool stop_at_newline) {
     bool had_physical_newline = false;
     while (!is_end(lex)) {
         if (is_physical_newline(lex)) {
-            if (stop_at_newline && lex->nested_bracket_level == 0) {
-                break;
+            if (lex->nested_bracket_level == 0) {
+                had_physical_newline = true;
+                if (stop_at_newline) {
+                    break;
```

Comment:
I think it's possible to just return true here, and false at the end of the function, and not need the `had_physical_newline` variable at all.

Then this if statement doesn't need to change, it can stay as a single expression with the &&.